src/analysis/retail/evoker/preservation/modules/talents/TitansGift.tsx

```diff
@@ -0,0 +1,170 @@
+  isCastFromBurst,
+} from '../../normalizers/EventLinking/helpers';
+
+class TitansGift extends Analyzer {
+  //Blossom
+  healingAddedToBlossoms: number = 0;
+  totalBlossomsCasted: number = 0;
+  buffedBlossoms: number = 0;
+  //Echo
+  healingAddedToEcho: number = 0;
+  totalEchoesCasted: number = 0;
+  buffedEchoes: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS_EVOKER.TITANS_GIFT_TALENT);
+
+    //Count total casts
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EMERALD_BLOSSOM_CAST),
+      this.countBlossomCasts,
+    );
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS_EVOKER.ECHO_TALENT),
+      this.countEchoCasts,
+    );
+
+    //Track blossom heals
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.EMERALD_BLOSSOM),
+      this.emeraldBlossomHeal,
+    );
+
+    //Track echo heals
+    this.addEventListener(
+      Events.heal
+        .by(SELECTED_PLAYER)
+        .spell([
+          SPELLS.DREAM_BREATH_ECHO,
+          SPELLS.EMERALD_BLOSSOM_ECHO,
+          SPELLS.LIVING_FLAME_HEAL,
+          SPELLS.SPIRITBLOOM_SPLIT,
+          SPELLS.SPIRITBLOOM_FONT,
+          SPELLS.SPIRITBLOOM,
+          SPELLS.VERDANT_EMBRACE_HEAL,
+        ]),
+      this.echoHeal,
+    );
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.REVERSION_ECHO, SPELLS.DREAM_BREATH_ECHO]),
+      this.echoHeal,
+    );
+
+    this.addEventListener(
+      Events.refreshbuff
+        .by(SELECTED_PLAYER)
+        .spell([SPELLS.REVERSION_ECHO, SPELLS.DREAM_BREATH_ECHO]),
+      this.echoHeal,
+    );
+  }
+
+  //Count total casts
+  countBlossomCasts() {
+    this.totalBlossomsCasted += 1;
+  }
+
+  countEchoCasts() {
+    this.totalEchoesCasted += 1;
+  }
+
+  //Track blossom healing added
+  emeraldBlossomHeal(event: HealEvent) {
+    const BlossomCast = getBlossomCast(event);
+    if (BlossomCast && isCastFromBurst(BlossomCast)) {
+      this.buffedBlossoms += 1;
+      const BlossomHeals = getHealEvents(event);
+      for (const BlossomHeal of BlossomHeals) {
+        this.healingAddedToBlossoms += (BlossomHeal.amount + (BlossomHeal.absorbed ?? 0)) / 5;
+      }
+    }
+  }
+
+  //Track echo healing added
+  echoHeal(event: HealEvent | ApplyBuffEvent | RefreshBuffEvent) {
+    const EchoApplication = getEchoAplication(event);
+    if (EchoApplication && isCastFromBurst(EchoApplication)) {
+      this.buffedEchoes += 1;
+      if (event.type === EventType.Heal) {
+        this.healingAddedToEcho += (event.amount + (event.absorbed ?? 0)) / 5;
+      } else {
+        if (event.ability.name === TALENTS_EVOKER.DREAM_BREATH_TALENT.name) {
+          const DbHealing = getDreamBreathHealing(event);
+          for (const DbHeal of DbHealing) {
+            this.healingAddedToEcho += (DbHeal.amount + (DbHeal.absorbed ?? 0)) / 5;
+          }
```

Comment:
This isn't strictly required but we generally try to use functional methods where possible
```suggestion
          this.healingAddedToEcho += dbHealing.reduce((prev, cur) => calculateEffectiveHealing(cur, TITANS_GIFT_INC) + prev, 0);
```